### PR TITLE
Extra helper functions for dllist

### PR DESCRIPTION
**Pull request by ashamis**
Adding extra helper function for dllist

### CODE REVIEW

File: src/ds/dllist.h [hunk 1]
```diff
@@ -66,6 +66,32 @@ namespace snmalloc
     T* tail = Terminator();
 
   public:
+    virtual ~DLList()
```
Comment (mjp41) [inline src/ds/dllist.h]: Why do you need the desctructor to be `virtual`?

Comment (ashamis) [reply src/ds/dllist.h]: I can remove that but the class is not marked final so theoretically someone could inherit from it.

Comment (mjp41) [reply src/ds/dllist.h]: I would prefer final and non-virtual.  You have increased the size of the allocator data structure with this change.  Every DLL now has a vtable, i.e. we have gone from 
```
      sizeof(DLList) == (sizeof(void*) * 2)
```
to
```
      sizeof(DLList) == (sizeof(void*) * 3)
```

Comment (ashamis) [reply src/ds/dllist.h]: done

**Pull request merged**